beamer/tests/contracts/test_request_manager.py

```diff
@@ -714,22 +742,41 @@ def test_withdraw_without_challenge_with_resolution(
         fill_id,
     )
 
-    # Register a L1 resolution
     contracts.messenger2.setLastSender(contracts.resolver.address)
+
+    if invalidate:
+        resolution_registry.invalidateFillHash(
+            request_id, fill_hash, chain.id, {"from": contracts.messenger2}
+        )
+    # Assert that invalidation works
+    assert resolution_registry.invalidFillHashes(fill_hash) == invalidate
+
+    # Register a L1 resolution
     resolution_registry.resolveRequest(
-        request_id, fill_hash, web3.eth.chain_id, claimer.address, {"from": contracts.messenger2}
+        request_id, fill_hash, web3.eth.chain_id, l1_filler, {"from": contracts.messenger2}
     )
 
+    # Assert that correct filler is resolved, it reverts the false invalidation
+    if invalidate and l1_filler == claimer:
+        assert not resolution_registry.invalidFillHashes(fill_hash)
```

Comment:
It would be nice to see the entire boolean table covered, e.g.
```
assert resolution_registry.invalidFillHashes(fill_hash) == f(invalidate, claimer, l1_filler)
assert resolution_registry.filllers(fill_hash) == g(invalidate, claimer, l1_filler)
```